crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -243,98 +230,57 @@ where
+// If successfull in getting the jwt token '71,xxxxx', the connection is established.
+fn check_device_status_c8y(tedge_config: &TEdgeConfig) -> Result<DeviceStatus, ConnectError> {
+    const C8Y_TOPIC_BUILTIN_JWT_TOKEN_DOWNSTREAM: &str = "c8y/s/dat";
+    const C8Y_TOPIC_BUILTIN_JWT_TOKEN_UPSTREAM: &str = "c8y/s/uat";
     const CLIENT_ID: &str = "check_connection_c8y";
-    const REGISTRATION_ERROR: &[u8] = b"41,100,Device already existing";
-    const DEVICE_TYPE: &str = "thin-edge.io";
 
-    let registration_payload = format!("100,{},{}", device_id, DEVICE_TYPE);
-
-    let mut options = MqttOptions::new(CLIENT_ID, DEFAULT_HOST, port);
+    let mut options = MqttOptions::new(
+        CLIENT_ID,
+        DEFAULT_HOST,
+        tedge_config.query(MqttPortSetting)?.into(),
+    );
     options.set_keep_alive(RESPONSE_TIMEOUT);
 
     let (mut client, mut connection) = rumqttc::Client::new(options, 10);
     let mut acknowledged = false;
 
-    client.subscribe(C8Y_TOPIC_ERROR_MESSAGE_DOWNSTREAM, AtLeastOnce)?;
+    client.subscribe(C8Y_TOPIC_BUILTIN_JWT_TOKEN_DOWNSTREAM, AtLeastOnce)?;
 
     for event in connection.iter() {
         match event {
             Ok(Event::Incoming(Packet::SubAck(_))) => {
                 // We are ready to get the response, hence send the request
-                client.publish(
-                    C8Y_TOPIC_BUILTIN_MESSAGE_UPSTREAM,
-                    AtLeastOnce,
-                    false,
-                    registration_payload.as_bytes(),
-                )?;
+                client.publish(C8Y_TOPIC_BUILTIN_JWT_TOKEN_UPSTREAM, AtLeastOnce, false, "")?;
             }
             Ok(Event::Incoming(Packet::PubAck(_))) => {
                 // The request has been sent
                 acknowledged = true;
             }
             Ok(Event::Incoming(Packet::Publish(response))) => {
                 // We got a response
-                if response.payload == REGISTRATION_ERROR {
+                let token = String::from_utf8(response.payload.to_vec()).unwrap();
+
+                if token.contains("71") {
                     return Ok(DeviceStatus::AlreadyExists);
+                } else {
+                    return Err(ConnectError::ConnectionCheckError);
                 }
             }
             Ok(Event::Outgoing(Outgoing::PingReq)) => {
                 // No messages have been received for a while
                 println!("Local MQTT publish has timed out.");
-                break;
+                return Err(ConnectError::ConnectionCheckError);
             }
             Ok(Event::Incoming(Incoming::Disconnect)) => {
                 eprintln!("ERROR: Disconnected");
-                break;
+                return Err(ConnectError::ConnectionCheckError);
             }
             Err(err) => {
                 eprintln!("ERROR: {:?}", err);
-                break;
+                return Err(ConnectError::ConnectionCheckError);
```

Comment:
Since all the `break` have been removed from the loop, the code after the loop seems dead.

Is the code following the loop dead now? I'm okay to get rid of the `DeviceStatus::Unknown` state but I would prefer to trigger a `ConnectError::TimeoutElapsedError`  error when appropriate. 

```
    if acknowledged {
        // The request has been sent but without a response
        Ok(DeviceStatus::Unknown)
    } else {
        // The request has not even been sent
        println!("\nMake sure mosquitto is running.");
        Err(ConnectError::TimeoutElapsedError)
    }
```

Reply:
done